proposal.html

```diff
@@ -523,6 +523,13 @@ <h1>(<emu-xref href="#sec-createdynamicfunction">19.2.1.1.1</emu-xref>) Runtime
 </emu-clause>
 
 
+<emu-clause id="proposal-sec-hosthassourcetextavailable">
+  <h1>HostHasSourceTextAvailable ( _func_ )</h1>
+  <p>HostHasSourceTextAvailable is an implementation-defined abstract operation that allows host environments to prevent the source text from being provided for a given function.</p>
+  <p>An implementation of HostHasSourceTextAvailable may complete normally or abruptly. Any abrupt completions will be propagated to its callers. The default implementation of HostHasSourceTextAvailable is to unconditionally return a normal completion with a value of *true*.</p>
```

Comment:
Not sure we need the flexibility to return abruptly, but it doesn't hurt anything.

Reply:
I figured since `toString` can already return abruptly today, it didn't hurt to allow the engine to control that as well.

Reply:
Allowing abrupt completions doesn't seem useful to me. `Function.prototype.toString` is currently spec'ed to either return a string or throw a TypeError. With this change, it either returns a string or throws any value, so the error case is no longer restricted to TypeError.

For example test262 tests will need to be written as:
```js
function f() {}

var errored = false;
var source;
try {
  source = f.toString();
} catch {
  // HostHasSourceTextAvailable is allowed to return abruptly, unconditionally pass in this case.
  errored = true;
}

if (!errored) {
  // NATIVE_FUNCTION_RE is defined in test262/harness/nativeFunctionMatcher.js
  assert(source === "function f() {}" || NATIVE_FUNCTION_RE.test(source));
}
```

If we don't allow abrupt completions, the test can be written as:
```js
function f() {}

var source = f.toString();
assert(source === "function f() {}" || NATIVE_FUNCTION_RE.test(source));
```

Nit: The result must be compared to `true`, so `HostHasSourceTextAvailable(func) is true, then return ...` instead of `HostHasSourceTextAvailable(func), then return ...`. And we should probably require HostHasSourceTextAvailable to return a boolean value. 

And I assume not requiring stable results is intentional? So HostHasSourceTextAvailable is allowed to return different results over time even when the same argument is passed to it?